### PR TITLE
Added some documentation regarding unit testing.

### PR DESCRIPTION
**Pull request by jonathanslenders**

### CODE REVIEW

**Comment by fshowalter** [comment]
This was a good hint, but I ended up with:

```
@pytest.fixture(autouse=True, scope="function")
def mock_input(mocker: MockFixture) -> PosixPipeInput:
    pipe_input = create_pipe_input()
    with create_app_session(input=pipe_input):
        yield pipe_input
    pipe_input.close()
```

This facilitates inputs to any application without having to hook into its constructor.

File: docs/pages/advanced_topics/unit_testing.rst [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+Without the :class:`~prompt_toolkit.output.DummyOutput`, all code that
+contained a :func:`~prompt_toolkit.shortcuts.print_formatted_text` call.
+call would fail in pytest, even if the purpose was to test the code around this
+call.
+
+Type checking
+-------------
+
+Prompt_toolkit 3.0 is fully type annotated. This means that if a
```
Comment (fshowalter) [inline docs/pages/advanced_topics/unit_testing.rst:81]: But it doesn't support PEP 561 (https://mypy.readthedocs.io/en/stable/installed_packages.html) so unless you follow imports you don't benefit.

Comment (jonathanslenders) [reply docs/pages/advanced_topics/unit_testing.rst:81]: Thanks for pointing out. This should fix that: https://github.com/prompt-toolkit/python-prompt-toolkit/pull/1117

**Comment by jonathanslenders** [comment]
@fshowalter : I have rewritten this documentation a bit, and included your example (hope that's fine).

**Pull request merged**